### PR TITLE
fix: correct race condition when concurrent pvcreate

### PR DESCRIPTION
**Pull request by jmclong**
Fix race conditions in concurrent volume provisioning
This PR addresses an issue where multiple simultaneous CreateVolume calls could lead to ResourceExhausted errors or volume groups missing physical volumes.

We've made three key improvements:

Better device detection - Now properly recognizing disks that already have LVM2_TYPE. Previously these were being filtered out during concurrent operations, causing inconsistent device selection.

Simplified physical volume creation - Implemented a more reliable get → create → get pattern that handles concurrent creation gracefully. If another process already created the PV, we now incorporate it rather than failing.

Smarter error handling - ResourceExhausted errors are now only returned when truly necessary: when no devices are available or when all PVs already belong to different volume groups.

The PR includes enhanced tests with an lsv4 (multi-disk) configuration and additional verification that volume groups contain all eligible disks and logical volumes have the correct stripe count.

AI Summary: 
This pull request introduces significant improvements to the handling of physical volumes (PVs) in the LVM controller, including more robust creation logic, better error handling, and updates to tests and workflows. The main focus is on ensuring PVs are correctly managed with respect to their volume groups, and that the system behaves predictably in edge cases.

Key changes include:

### LVM Controller Logic Improvements

* Refactored `EnsurePhysicalVolumes` in `lvm.go` to take the volume group name as an argument, and reworked the logic to:
  - Ensure PVs are only included if they are not already part of another VG.
  - Skip devices already in the correct VG.
  - Fail with a resource exhausted error if no suitable disks are found.
  - Extracted device creation logic into a new `ensurePhysicalVolume` helper function for clarity and testability. [[1]](diffhunk://#diff-b73642e31569cb1a69dbe6249611ea6a220756baed47163c1a7010d6f3e595afL311-R308) [[2]](diffhunk://#diff-b73642e31569cb1a69dbe6249611ea6a220756baed47163c1a7010d6f3e595afL330-R388)
* Removed the special case for `ErrPVAlreadyInVolumeGroup` in `EnsureVolumeGroup`, now handled earlier in PV creation.
* Updated `AvailableCapacity` and related calls to match the new probe interface signature.

### Testing Enhancements

* Updated all relevant tests to use the new `EnsurePhysicalVolumes(ctx, vgName)` signature and logic, including more realistic scenarios for PVs already in use, in the correct VG, or in another VG. [[1]](diffhunk://#diff-4c6592132c9ffe4d7605f48529cf585d714459a6a3f6b9a3873f97d7f1922c1cR198-R308) [[2]](diffhunk://#diff-4c6592132c9ffe4d7605f48529cf585d714459a6a3f6b9a3873f97d7f1922c1cL329-R329)
* Expanded test coverage for error and edge cases in PV creation, including concurrent creation and PVs already existing.
* Removed the now-unused `createPvs` helper function.
* Updated all mocks to match the new probe and LVM manager interfaces. [[1]](diffhunk://#diff-5060a8629f8bb63873dee9181ae9568fa1ab4fdcd4bbc7596e9ef1d14089cc19L331-R331) [[2]](diffhunk://#diff-5060a8629f8bb63873dee9181ae9568fa1ab4fdcd4bbc7596e9ef1d14089cc19L342-R342) [[3]](diffhunk://#diff-5060a8629f8bb63873dee9181ae9568fa1ab4fdcd4bbc7596e9ef1d14089cc19L373-R373) [[4]](diffhunk://#diff-5060a8629f8bb63873dee9181ae9568fa1ab4fdcd4bbc7596e9ef1d14089cc19L397-R397) [[5]](diffhunk://#diff-5060a8629f8bb63873dee9181ae9568fa1ab4fdcd4bbc7596e9ef1d14089cc19L417-R417) [[6]](diffhunk://#diff-4c6592132c9ffe4d7605f48529cf585d714459a6a3f6b9a3873f97d7f1922c1cL609-R609) [[7]](diffhunk://#diff-4c6592132c9ffe4d7605f48529cf585d714459a6a3f6b9a3873f97d7f1922c1cL620-R627)

### Workflow and Deployment Updates

* Added a new integration test job for the "nvme-v4" AKS template in the GitHub Actions workflow.
* Added deployment parameters for the "nvme-v4" AKS template.

These changes collectively improve the reliability and maintainability of PV management in the LVM controller, ensuring correct behavior in a wider range of real-world scenarios.

### CODE REVIEW

File: internal/pkg/probe/probe.go [hunk 1]
```diff
@@ -36,32 +36,40 @@ func New(b block.Interface, f *Filter) Interface {
...
 
-	var unformatted []block.Device
+	var availableDevices []block.Device
 	for _, device := range devices.Devices {
 		if !m.filter.Match(device) {
 			log.V(3).Info("device filtered out", "device", device)
 			continue
 		}
-		isFormatted, err := m.IsFormatted(device.Path)
+		fsType, isFormatted, err := m.IsFormatted(device.Path)
```
Comment (croomes) [inline internal/pkg/probe/probe.go:52]: I think I'd create a new `IsLVM2(path string) (bool, err)` to hide the fsType
check. Then `IsFormatted()` doesn't need to return the fstype. Both can be
wrappers for a shared func in block.go.

File: internal/pkg/probe/probe.go [hunk 2]
```diff
@@ -7,9 +7,9 @@ import (
 	"context"
 	"fmt"
 
-	"github.com/go-logr/logr"
-
 	"local-csi-driver/internal/pkg/block"
+
+	"sigs.k8s.io/controller-runtime/pkg/log"
```
Comment (croomes) [inline internal/pkg/probe/probe.go:12]: nit: import ordering

**Review by croomes — APPROVED** [review]
Thanks for the fixes - looks great!
